src/NLog/LogFactory.cs

```diff
@@ -1166,6 +1168,42 @@ public LogFactory LoadConfiguration(string configFile)
             return LoadConfiguration(configFile, optional: false);
         }
 
+        /// <summary>
+        /// Reload the current logging configuration and apply it.
+        /// </summary>
+        /// <returns>LogFactory instance for fluent interface</returns>
+        public LogFactory ReloadConfiguration()
+        {
+            InternalLogger.Info("Reloading configuration...");
+            var errorRaised = false;
+
+            if (_config != null)
```

Comment:
When `_config == null` then you should should just reference the `Configuration`-property to perform default load of NLog.config.

Reply:
Too confusing and tricky now. Don't really get it why it's needs to be so complicated...

Reply:
(well yes, I understand, the getters and setters are too complicated)

Reply:
Anyway, dunno what I should do now

Reply:
Something like this:

```c#
public LogFactory ReloadConfiguration()
{
  if (_config != null)
  {
       Configuration = _config.ReloadNewConfig();
       NotifyConfigurationReloaded();
  }
  else
  {
       _config = Configuration;
       if (_config != null)
       {
             NotifyConfigurationReloaded();
       }
  }
  return this;
}
```

Reply:
Done, I also added a try catch. 

Also I did a refactor of the property, as it's really confusing to me that it's doing so much. So made that clear with separate methods. Let me know if you like it. Otherwise I will just remove that commit.

Reply:
Think it is a good idea to extract logic `Configuration`-property into `ApplyNewConfig`. But I think it is a bad idea to change all places to call `ApplyNewConfig` directly. Instead all should just continue to assign the `Configuration`-property. (Only `Configuration`-property should call `ApplyNewConfig`)